### PR TITLE
Fixes holo switchtool item attack animation

### PR DESCRIPTION
**Pull request by jknpj**
muh immulshions

They would simply show the item inside the tool in their item attack animation before, muh immulsions.

PS:. also cleans its code a bit.

:cl:
 * bugfix: Fixes holo switchtool item attack animation

### CODE REVIEW

**Comment by DrSnips** [comment]
Was this issue only with the holotool or was it with all switchtools?

File: code/game/objects/items/weapons/switchtool.dm [hunk 1]
```diff
@@ -355,6 +355,8 @@
 		item_state = "uvsoap"
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/switchtool.dm]: presumably these are the ```item_states``` of the deployed item.
so why isn't this huge block of ifs just:
```
if(deployed)
    item_state = deployed.item_state
else
    item_state = "Hswitchtool"
```

**Comment by DrSnips** [comment]
I didn't really care to do a lot of cleanup in the code and just stuck with what worked at the time. If you do make a change like that, the sprite names do need to match case in the dmi and code to work correctly.

**Comment by DrSnips** [comment]
Not that it should affect the recent changes because it was inhands, but the item sprite overlays for the tools use the case sensitive aspect for the names to use the holotool kinds over the normal switchtool overlay sprites. (Multitool vs multitool).
I planned to seperate them into seperate dmis later because I planned to make them multicolor if it got merged, but since there has been prs open constantly to fix my shitcode, I cant really do much to the switchtools code for multicolor. Although I'm glad that the code is being cleaned up better than I could have done.

**Comment by Riekland** [comment]
we could fix this easily by removing attack animations

**Comment by jknpj** [comment]
@DrSnips nailed it.

Wait, i fucked up.

File: code/game/objects/items/weapons/switchtool.dm [hunk 2]
```diff
@@ -315,46 +312,12 @@
...
-	else if(istype(deployed, /obj/item/weapon/melee/energy/hfmachete/activated))
-		item_state = "sharper sword"
-	else if(istype(deployed, /obj/item/weapon/soap/holo))
-		item_state = "uvsoap"
+	if(deployed)
+		item_state = "holo_[initial(deployed.icon_state)]"
+		deployed.appearance = appearance
 	else
-		item_state = "Hswitchtool"
+		item_state = initial(icon_state)
```
Comment (Exxion) [inline code/game/objects/items/weapons/switchtool.dm:320]: I believe this should read `initial(item_state)`

Comment (Exxion) [reply code/game/objects/items/weapons/switchtool.dm:320]: OK I see what you're doing now and I don't really like it but I guess it works

**Comment by Exxion** [comment]
Sorry the second comment I added was to say that *was* actually the right way to do it, how you have it set up
I think, at least

**Pull request merged**